mlflow/pyspark/ml/__init__.py

```diff
@@ -1003,7 +1003,7 @@ def _log_posttraining_metadata(estimator, spark_model, params, input_df):
                 spark = SparkSession.builder.getOrCreate()
 
                 def _get_input_example_as_pd_df():
-                    feature_cols = list(get_feature_cols(input_df.schema, spark_model))
+                    feature_cols = list(get_feature_cols(input_df, spark_model))
                     limited_input_df = input_df.select(feature_cols).limit(
```

Comment:
We should add a `limited_input_df.cache()` here, otherwise each time we running spark jobs on limited_input_df it will trigger upstream spark dataframe (e.g. the `input_df`) to be recomputed.

Reply:
With a cache deletion after the execution of the `.predict()` as well?